Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -190,6 +169,64 @@ public static string GetLanguageVersion(string targetFramework)
             return Unknown();
         }
 
+        // Knowledge on unique machine Ids for different OSes obtained from https://github.com/denisbrodbeck/machineid
+        public static string GetAnonymizedUserId()
+        {
+            var id = string.Empty;
+            var currentOs = System.Environment.OSVersion.Platform;
+            Match match = null;
```

Comment:
Rather than creating this `Match` variable here, let's have each case populate the id. We're not really saving any lines of code and the code duplication is rather minimal.